proxy/Main.cc

```diff
@@ -1943,17 +1943,28 @@ main(int /* argc ATS_UNUSED */, const char **argv)
       int delay_p = 0;
       REC_ReadConfigInteger(delay_p, "proxy.config.http.wait_for_cache");
 
+      // Use a condition variable to check if we are ready to call
+      // start_HttpProxyServer() when num_accept_threads is set to 0.
+      {
+        std::unique_lock<std::mutex> lock(proxyServerMutex);
+        proxyServerCheck.wait(lock, [] { return et_net_threads_ready; });
+      }
+
       // Delay only if config value set and flag value is zero
       // (-1 => cache already initialized)
       if (delay_p && ink_atomic_cas(&delay_listen_for_cache_p, 0, 1)) {
         Debug("http_listen", "Delaying listen, waiting for cache initialization");
       } else {
-        // Use a condition variable to check if we are ready to call
-        // start_HttpProxyServer() when num_accept_threads is set to 0.
-        std::unique_lock<std::mutex> lock(proxyServerMutex);
-        proxyServerCheck.wait(lock, [] { return et_net_threads_ready; });
         start_HttpProxyServer(); // PORTS_READY_HOOK called from in here
       }
+
+      // Start the back door, since it's just a special HttpProxyServer,
+      // the requirements to start it has been met if we got here.
```

Comment:
Add some comment about this making the `traffic_cop` healthchecks work before the cache is ready in some cases.

Reply:
`requirements` != `has` - number your verb correctly!